lib/src/models/subreddit_moderation.dart

```diff
@@ -0,0 +1,647 @@
+    }
+    return ListingGenerator.generator(
+        _subreddit.reddit, _formatApiPath('about_log'),
+        limit: limit, params: params);
+  }
+
+  /// Returns a [Stream<UserContent>] of [Comment]s and [Submission]s in the
+  /// mod queue.
+  ///
+  /// If `only` is provided, only [Comment]s or [Submission]s will be returned
+  /// exclusively.
+  Stream<UserContent> modQueue(
+          {SubredditModerationContentTypeFilter only, int limit}) =>
+      _subredditModerationListingGeneratorHelper('about_modqueue', only,
+          limit: limit);
+
+  /// Returns a [Stream<UserContent>] of [Comment]s and [Submission]s which
+  /// have been reported.
+  ///
+  /// If `only` is provided, only [Comment]s or [Submission]s will be returned
+  /// exclusively.
+  Stream<UserContent> reports(
+          {SubredditModerationContentTypeFilter only, int limit}) =>
+      _subredditModerationListingGeneratorHelper('about_reports', only,
+          limit: limit);
+
+  /// Returns the current settings for the [Subreddit].
+  Future<SubredditSettings> settings() async {
+    final data = (await _subreddit.reddit
+        .get(_formatApiPath('subreddit_settings'), objectify: false))['data'];
+    return new SubredditSettings._(_subreddit, data);
+  }
+
+  /// Returns a [Stream<UserContent>] of [Comment]s and [Submission]s which
+  /// have been marked as spam.
+  ///
+  /// If `only` is provided, only [Comment]s or [Submission]s will be returned
+  /// exclusively.
+  Stream<UserContent> spam(
+          {SubredditModerationContentTypeFilter only, int limit}) =>
+      _subredditModerationListingGeneratorHelper('about_spam', only,
+          limit: limit);
+
+  /// Returns a [Stream<UserContent>] of unmoderated [Comment]s and
+  /// [Submission]s.
+  Stream<UserContent> unmoderated({int limit}) =>
+      _subredditModerationListingGeneratorHelper('about_unmoderated', null,
+          limit: limit);
+
+  /// Returns a [Stream<Message>] of unread moderator messages.
+  Stream<Message> unread({int limit}) =>
+      _subredditModerationListingGeneratorHelper('moderator_unread', null,
+          limit: limit);
+
+  /// Update the [Subreddit]s settings.
+  Future update(SubredditSettings updated) async {
+    if (updated == null) {
+      throw new DRAWArgumentError("Field 'updated' cannot be null.");
+    }
+    final data = new Map.from(updated._data);
+    final remap = {
+      'allow_top': 'default_set',
+      'lang': 'language',
+      'link_type': 'content_options',
+      'type': 'subreddit_type',
+      'sr': 'subreddit_id',
+    };
+
+    // Remap keys to what Reddit expects (this is dumb on their part).
+    remap.forEach((k, v) {
+      if (data[v] != null) {
+        data[k] = data[v];
+        data.remove(v);
+      }
+    });
+
+    // Not a valid key for the response.
+    if (data.containsKey('domain')) {
+      data.remove('domain');
+    }
+
+    // Cleanup the map before we send the request.
+    data.forEach((k, v) {
```

Comment:
Null-Aware Operators, why don't you use em, they are beautiful! 😄 

I had to check: (Link: http://blog.sethladd.com/2015/07/null-aware-operators-in-dart.html)

But from the link:
If _[Object]_ is null, the entire expression returns null. Otherwise, _[method]_ is called and returned.

`data.forEach((k, v) -> data[k] = v?.toString());`

Reply:
Notice that if `v == null` we set `data[k] = 'null'`. If we don't explicitly replace `null` with `'null'`, we end up throwing an exception in the HTTP library since `null is String == false`.

Reply:
I'm an idiot, my bad!